tests/formik/ActionBlock.test.js

```diff
@@ -0,0 +1,110 @@
+import React from "react";
+import { render, screen, waitFor } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import { Formik, Form } from "formik";
+
+import { ActionBlock, Input } from "../../lib/components/formik";
+
+const TestActionBlock = ({
+  onSubmit = () => {},
+  submitButtonProps = {},
+  cancelButtonProps = {},
+}) => {
+  const handleSubmit = (value) => {
+    onSubmit(value);
+  };
+  return (
+    <Formik
+      initialValues={{
+        input: "",
+      }}
+      onSubmit={handleSubmit}
+    >
+      <Form>
+        <Input label="Input label" name="input" />
+        <ActionBlock
+          submitButtonProps={submitButtonProps}
+          cancelButtonProps={cancelButtonProps}
+        />
+      </Form>
+    </Formik>
+  );
+};
+
+describe("formik/ActionBlock", () => {
+  it("should render without any errors", () => {
+    render(<TestActionBlock />);
+
+    expect(screen.getByText(/Save changes/i)).toBeInTheDocument();
+    expect(screen.getByText(/Cancel/i)).toBeInTheDocument();
+  });
+
+  it("should apply the classNames correctly", () => {
+    const { container } = render(
+      <Formik>
+        <Form>
+          <ActionBlock className="test-class" />
+        </Form>
+      </Formik>
+    );
+
+    expect(container.querySelector(".test-class")).toBeInTheDocument();
+  });
+
+  it("should apply the button props correctly", () => {
+    render(
+      <TestActionBlock
+        cancelButtonProps={{ label: "cancel" }}
+        submitButtonProps={{ label: "submit" }}
+      />
+    );
+
+    expect(screen.getByRole("button", { name: "cancel" })).toBeInTheDocument();
+    expect(screen.getByRole("button", { name: "submit" })).toBeInTheDocument();
+  });
+
+  it("should render the submit button disabled if the form is empty", () => {
+    render(<TestActionBlock />);
+
+    const submitButton = screen.getByRole("button", { name: /Save changes/i });
+    expect(submitButton).toBeDisabled();
+
+    const input = screen.getByRole("textbox");
+    userEvent.type(input, "test");
+    expect(submitButton).not.toBeDisabled();
```

Comment:
I think this part can be removed from here, since we are testing testing the same in the next test